components/places/src/api/matcher.rs

```diff
@@ -199,15 +197,15 @@ pub struct SearchResult {
 
     /// The URL to open when the user confirms a match. This is
     /// equivalent to `nsIAutoCompleteResult.getFinalCompleteValueAt`.
-    pub url: Url,
+    pub url: String,
```

Comment:
UniFFI doesn't know about `Url`, so I had to make these `String` in order to expose them to the foreign language. We could have a little `From`/`Into` layer here that converted between the public string-using version and an internal Url-using version if we wanted to.

Reply:
autofill would (maybe) like this too - there it would be more about 'guid' where we use a guid type internally but strings with uniffi, which jars a little. I haven't looked into how hard that would be though, but just a little +1 for that concept.

Reply:
So it was this chain of comments that distracted me and ended up with me opening https://github.com/mozilla/uniffi-rs/issues/475. The fact that autofill and logins ended up changing guids to strings is somewhat justifiable, but for our places component to start internally using Strings instead of Urls bothers me alot more.